### ISSUE TITLE
Library Conversion: TF Ranking

### ISSUE THREAD

**Issue by dynamicwebpaige** [tensorflow/tensorflow#25445]
[**TensorFlow Ranking**](https://github.com/tensorflow/ranking) is a library for Learning-to-Rank (LTR) techniques on the TensorFlow platform. It contains the following components:

*   Commonly used loss functions including pointwise, pairwise, and listwise losses.
*   Commonly used ranking metrics like Mean Reciprocal Rank (MRR) and Normalized Discounted Cumulative Gain (NDCG).
*   [Multi-item (also known as groupwise) scoring functions](https://arxiv.org/abs/1811.04415).
*   [LambdaLoss](https://ai.google/research/pubs/pub47258) implementation for
    direct ranking metric optimization.
*   [Unbiased Learning-to-Rank](http://www.cs.cornell.edu/people/tj/publications/joachims_etal_17a.pdf)
    from biased feedback data.

The objective of this issue is to migrate TensorFlow Ranking to TF 2.0, with support for Keras models and Ranking Estimator. The key objectives are:

1. Removing [deprecated TensorFlow functions](https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md).
2. Migrate to newer version of [feature columns](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/feature_column). `tfr.feature.encode_features` is probably no longer needed.
3. Shift from `tflearn` to `estimator`, and from `experiment_gn` to `train_and_evaluate`.
4. Support for model definition via Keras: Keras allows for object oriented definition of models, in contrast with function closures currently used in TF Ranking.
5. Support Ranking Estimator as a canned estimator.

**Comment by Rajiv2605** [comment]
@dynamicwebpaige I'm interested to work on this issue! It would be great if you could provide a starting point :)

**Comment by Tahsin-Mayeesha** [comment]
Would it be ok to make it a GSOC proposal? @dynamicwebpaige

**Comment by eggie5** [comment]
TF Ranking already uses Estimators and `feature_column`s

**Comment by Tahsin-Mayeesha** [comment]
Is there room for doing 4 and 5?

**Comment by Tahsin-Mayeesha** [comment]
@eggie5 Yeah, given that, any idea how can I customize the project proposal for the other points? :)

**Comment by ramakumar1729** [comment]
TensorFlow Ranking  is now fully compatible with TF 2.0 RC0. See release [notes](https://github.com/tensorflow/ranking/releases/tag/v0.1.4) for v0.1.4.

**Comment by ramakumar1729** [comment]
Closing this issue, as the library is TF2.0 compatible. Ranking models using Keras are tracked in a separate [issue](https://github.com/tensorflow/ranking/issues/32).
